src/glados/tui.py

```
@@ -397,88 +397,143 @@ async def action_quit(self) -> None:
         try:
             app = cls()
             app.run()
         except KeyboardInterrupt:
             logger.info("Application interrupted by user. Exiting.")
-            if app is not None and hasattr(app, "action_quit") and callable(app.action_quit):
-                app.action_quit()  # This will now call the improved action_quit
-            # No explicit sys.exit(0) here; Textual's app.exit() will handle it.
+            if app is not None and hasattr(app, "action_quit"):
+                import asyncio
+                try:
+                    asyncio.create_task(app.action_quit())
```

Ruff (0.11.9)

Store a reference to the return value of `asyncio.create_task`

(RUF006)